nntrainer/layers/zoneout_lstmcell.cpp

```diff
@@ -189,31 +193,39 @@ void ZoneoutLSTMCellLayer::finalize(InitLayerContext &context) {
     lstm_cell_state_dim, "lstm_cell_state", Tensor::Initializer::NONE, true,
     TensorLifespan::ITERATION_LIFESPAN);
 
-  // hidden_state_zoneout_mask_dim = [ max_timestep * batch_size, 1, 1, unit ]
-  const TensorDim hidden_state_zoneout_mask_dim(max_timestep * batch_size, 1, 1,
-                                                unit);
-  if (test) {
-    wt_idx[ZoneoutLSTMParams::hidden_state_zoneout_mask] =
-      context.requestWeight(hidden_state_zoneout_mask_dim,
-                            Tensor::Initializer::NONE, WeightRegularizer::NONE,
-                            1.0f, "hidden_state_zoneout_mask", false);
-  } else {
-    wt_idx[ZoneoutLSTMParams::hidden_state_zoneout_mask] =
-      context.requestTensor(
-        hidden_state_zoneout_mask_dim, "hidden_state_zoneout_mask",
-        Tensor::Initializer::NONE, false, TensorLifespan::ITERATION_LIFESPAN);
+  if (hidden_state_zoneout_rate > epsilon) {
+    // hidden_state_zoneout_mask_dim = [ max_timestep * batch_size, 1, 1, unit ]
+    const TensorDim hidden_state_zoneout_mask_dim(max_timestep * batch_size, 1,
```

Comment:
I think this should be better off that each cell has it's own zoneout mask instead of sharing.

Reply:
It will be done in the following pr